src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/TransitLayerUpdater.java

```diff
@@ -43,22 +46,25 @@
                 tripPatternForDateMapCache = new HashMap<>();
 
   public TransitLayerUpdater(
-      TransitLayer transitLayer,
+      Graph graph,
       Map<ServiceDate, TIntSet> serviceCodesRunningForDate
   ) {
-    this.transitLayer = transitLayer;
+    this.graph = graph;
     this.serviceCodesRunningForDate = serviceCodesRunningForDate;
   }
 
   public void update(Set<Timetable> updatedTimetables) {
-    if (transitLayer == null) { return; }
+    // Make a shallow copy of the realtime transit layer
+    TransitLayer realtimeTransitLayer = new TransitLayer(graph.realtimeTransitLayer);
```

Comment:
Personally I would not make the shallow-copy method a constructor because it is tailored for a specific use case. I'd make a basic private all-fields constructor that does not transform any of its input parameters, then call that constructor from a factory method that makes a shallow copy of the one Map tripPatternsForDate. Then that factory method is specific to the use case, and can have a name like TransitLayer#copyForTripPatternUpdates. This is of course not critical now, since all constructor calls are for this purpose. It's more of a long-term style and maintenance issue.

Reply:
Actually, it is also used for making a shallow copy of the non-realtime TransitLayer to initialize the RealtimeTransitLayer. As you said, we could probably merge this now and rethink how that should work.